### PR TITLE
Switch APIs to the appropriate packages

### PR DESCRIPTION
**Pull request by gabrielwen**
Related: #68 #73 
Forked from: #84 #85 

- Remove `gcp_plugin.go` and `aws_plugin.go`
- Migrate all the codes to use plugin definitions in `/pkg/apis/apps/plugins`

```
# Package structure
/pkg ----> /apis/apps/{kfdef, plugins} # external facing APIs
     |
     ----> /kfconfig/{types.go, gcp_plugin.go, aws_plugin.go} # internal data structures for /kfapps and /kfupgrade

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfctl/93)

### CODE REVIEW

File: pkg/kfapp/gcp/gcp_test.go [hunk 1]
```diff
@@ -6,6 +6,8 @@ import (
 	"github.com/gogo/protobuf/proto"
 	kftypes "github.com/kubeflow/kfctl/v3/pkg/apis/apps"
 	"github.com/kubeflow/kfctl/v3/pkg/apis/apps/kfconfig"
+	kfdeftypes "github.com/kubeflow/kfctl/v3/pkg/apis/apps/kfdef/v1beta1"
```
Comment (jlewi) [inline pkg/kfapp/gcp/gcp_test.go]: Why are you using the public apis here?

**Comment by jlewi** [comment]
@gabrielwen Which directory contains the external/public versioned APIs/structs? And which directory contains the internal versions like KFConfig?

I thought
pkg/apis/...
https://github.com/kubeflow/kfctl/tree/master/pkg/apis

contained the external versions. So I would have thought that all of the internal code e.g. coordinator and the code for the plugins would be using the internal structs

Where are the internal structs?

Does kubeflow/kfctl#84 need to be merged first?

**Comment by gabrielwen** [comment]
> @gabrielwen Which directory contains the external/public versioned APIs/structs? And which directory contains the internal versions like KFConfig?
> 
> I thought
> pkg/apis/...
> https://github.com/kubeflow/kfctl/tree/master/pkg/apis
> 
> contained the external versions. So I would have thought that all of the internal code e.g. coordinator and the code for the plugins would be using the internal structs
> 
> Where are the internal structs?
> 
> Does #84 need to be merged first?

@jlewi discussed offline I've copied plugins to /kfconfig in #85 
Package structure was updated in description.  Could you help approve #84 and #85 so that this PR can be unblocked?

**Comment by jlewi** [comment]
@gabrielwen looks like #84 and #85 have been approved and are in the merge pool.

So after #84 and #85 are submitted is the idea that the kfapps (coordinator.go and plugin code) will only use the internal structures in kfconfig?

**Comment by gabrielwen** [comment]
> @gabrielwen looks like #84 and #85 have been approved and are in the merge pool.
> 
> So after #84 and #85 are submitted is the idea that the kfapps (coordinator.go and plugin code) will only use the internal structures in kfconfig?

yes.  making changes to make sure of it.  will let you know when this is ready.

/hold

**Pull request merged**